File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.java
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
        HashMap<Character, Integer>  mp = new HashMap<>();
        int count=words.length;
        for(int i=0; i<allowed.length();i++){
            mp.put(allowed.charAt(i),1);
        }
        for(int i=0; i<words.length;i++){
            for(int j=0; j<words[i].length();j++){
                if(mp.get(words[i].charAt(j))==null){
                    count--;
                    break;
                }
            }
        }
        return count;
    }
}

File: 1684-count-the-number-of-consistent-strings/1684-count-the-number-of-consistent-strings.cpp
class Solution {
public:
    int countConsistentStrings(string allowed, vector<string>& words) {
        int result=0;
        unordered_set<char> s(allowed.begin(), allowed.end());
        
        for(int i=0; i<words.size();i++)
        {
            bool flag=false;
            for(int j=0; j<words[i].size();j++)
            {
                if(s.find(words[i][j])==s.end())
                {
                    flag=true;
                    break;
                }
            }
            if(!flag)
            {
                result++;
            }
        }
        
        return result;
    }
};